File: IT/Oggetti/C++/multipleInheritance.cpp
/* file multipleInheritance.cpp */

#include <iostream>

using std::cout;
using std::endl;
using std::boolalpha;

class Animal 
{
	public:
		void walk();
};

void Animal::walk()
{
	cout << "Sto passeggiando." << endl;
}

class Flyer 
{
	public:
		void takeFlight();
};

void Flyer::takeFlight()
{
	cout << "Sto decollando!" << endl;
}

/* la classe Person estende le classi Animal e Flyer */

class Person : public Animal, public Flyer 
{
};

int main()
{
	Person *p = new Person();

	cout << "p è un animale? " << boolalpha << (bool)dynamic_cast<Animal*>(p) << endl << "p è un pilota di aereo? " << (bool)dynamic_cast<Flyer*>(p) << endl;

	/* invoca le funzioni membro ereditate da entrambe le superclassi */

	p->walk();
	p->takeFlight();

	delete p;
}

File: IT/Oggetti/Python/multipleInheritance.py
# file multipleInheritance.py
# -*- coding: utf-8 -*-

from __future__ import print_function

class Animal:
	def walk(self):
		print("Sto passeggiando.")

class Flyer:
	def takeFlight(self):
		print("Sto decollando!")

# la classe Person estende le classi Animal e Flyer
class Person(Animal, Flyer):
	pass

p = Person()

print("p è un animale? {}".format(isinstance(p, Animal)))
print("p è un pilota di aereo? {}".format(isinstance(p, Flyer)))

# invoca le funzioni membro ereditate da entrambe le superclassi

p.walk()
p.takeFlight()

